src/controllers/trigger/common/trigger.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { internalServerError, notFound, ok } from '@adobe/spacecat-shared-http-utils';
+
+import { isAuditForAllUrls, isAuditForAllDeliveryTypes, sendAuditMessages } from '../../../support/utils.js';
+
+/**
+ * Retrieves sites for auditing based on the input URL. If the input URL has the value
+ * 'all', then all sites will be audited. Otherwise, the input URL is assumed to be a base URL.
+ * Sites are filtered to only include sites that have not disabled audits.
+ *
+ * @param {Object} dataAccess - The data access object for site operations.
+ * @param {string} url - The URL to check for auditing.
+ * @returns {Promise<Array<Site>>} The sites to audit.
+ * @throws {Error} Throws an error if the site is not found.
+ */
+async function getSitesToAudit(dataAccess, url, deliveryType) {
```

Comment:
This could be made more readable like this:
```suggestion
async function getSitesToAudit(dataAccess, url, deliveryType) {
  let sitesToAudit;

  if (isAuditForAllUrls(url)) {
    sitesToAudit = isAuditForAllDeliveryTypes(deliveryType) 
      ? await dataAccess.getSites() 
      : await dataAccess.getSitesByDeliveryType(deliveryType);
  } else {
    const site = await dataAccess.getSiteByBaseURL(url);
    sitesToAudit = site ? [site] : [];
  }

  return sitesToAudit.filter(site => !site.getAuditConfig().auditsDisabled());
}
```